covsirphy/cleaning/pcr_data.py

```
@@ -0,0 +1,439 @@
+                        - Tests (int): the number of total tests performed
+                        - Confirmed(int): the number of confirmed cases
+                str/bool: kind of complement or False
+        """
+        country_alias = self.ensure_country_name(country)
+        subset_arg_dict = {
+            "country": country, "province": province,
+            "start_date": start_date, "end_date": end_date,
+        }
+        try:
+            return (self.subset(**subset_arg_dict), False)
+        except ValueError:
+            raise SubsetNotFoundError(
+                country=country, country_alias=country_alias, province=province,
+                start_date=start_date, end_date=end_date) from None
+
+    def positive_rate(self, country, province=None, window=3, show_figure=True, filename=None):
+        """
+        Return the PCR rate of a country as a dataframe.
+
+        Args:
+            country(str): country name or ISO3 code
+            window (int): window of moving average, >= 1
+            show_figure (bool): if True, show the records as a line-plot.
+            filename (str): filename of the figure, or None (show figure)
+
+        Returns:
+            pandas.DataFrame
+                Index:
+                    reset index
+                Columns:
+                    - Date (pd.TimeStamp): Observation date
+                    - Tests (int): the number of total tests performed
+                    - Confirmed (int): the number of confirmed cases
+                    - PCR_Rate (float): positive rate (%) of the daily cases over the total daily tests performed
+
+        Notes:
+            If non monotonic records were found for either cases or tests,
+            "(complemented)" will be added to the title of the figure.
+        """
+        window = self.ensure_natural_int(window, name="window")
+        df = self.records(country, province)[0]
+        if df[self.TESTS].max() == 0:
```

Please raise `SubsetNotFoundError` rather than stdout warnings.

---

` == 0` is not required in if-statement because checking 0 or not.

---

Why `== 0` is not required? You mean by adding `if not`, right?

---

You are right, `if not df[self.TESTS].max()` is required.